powermon/outputs/abstractoutput.py

```diff
@@ -1,43 +1,51 @@
+""" outputs / abstractoutput.py """
 import logging
 from abc import ABC, abstractmethod
-from enum import auto
 
-from strenum import LowercaseStrEnum
-
-# from powermon.dto.outputDTO import OutputDTO
-from powermon.formats.abstractformat import AbstractFormat
-from powermon.libs.mqttbroker import MqttBroker
+# from powermon.libs.mqttbroker import MqttBroker
 from powermon.commands.result import Result
+from powermon.dto.outputDTO import OutputDTO
+from powermon.formats.abstractformat import AbstractFormat
 
 log = logging.getLogger("Output")
 
-class OutputType(LowercaseStrEnum):
-    API_MQTT = auto()
-    MQTT = auto()
-    SCREEN = auto()
 
 class AbstractOutput(ABC):
+    """ base class for all output modules """
+    def __init__(self, name=None) -> None:
+        self.name = name
+        self.command_code : str = "not_set"
+        self.device_id : str = "not_set"
+        self.topic = None
+        self.formatter = None
 
     def set_formatter(self, formatter : AbstractFormat):
+        """ set the formatter for this output """
         self.formatter = formatter
 
-    def get_topic(self) -> str:
-        return ""
-
     @abstractmethod
-    def process(self, result: Result):
-        pass
-
-    def set_mqtt_broker(self, mqtt_broker: MqttBroker):
-        pass
+    def process(self, result: Result, mqtt_broker=None):
```

Comment:
changed this from device to mqtt_broker, easier to test and the same logic.

Reply:
for hass we use the device details to populate the 'device' for the mqtt messages
i was debating passing device and mqtt_broker separately or adding mqtt_broker to device, not really happy with either, but need both somehow for certain outputs

Reply:
hass is a format not an output, it already has device.

Reply:
ok great

Reply:
> hass is a format not an output, it already has device.

how?